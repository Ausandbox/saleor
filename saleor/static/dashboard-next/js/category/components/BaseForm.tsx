import Button from "material-ui/Button";
import Card, { CardContent, CardActions } from "material-ui/Card";
import Typography from "material-ui/Typography";
import { withStyles, WithStyles } from "material-ui/styles";
import * as React from "react";
import { Component } from "react";

import { TextField } from "../../components/TextField";
import i18n from "../../i18n";

const decorate = withStyles(theme => ({
  cardActions: {
    flexDirection: "row-reverse" as "row-reverse"
  },
  textField: {
    marginBottom: "2rem"
  }
}));

interface BaseCategoryFormProps {
  confirmButtonLabel: string;
  description: string;
  handleConfirm(data);
  name: string;
  title: string;
  errors: Array<{
    field: string;
    message: string;
  }>;
}

interface BaseCategoryFormState {
  formData: {
    description: string;
    name: string;
  };
}

export const BaseCategoryForm = decorate(
  class BaseCategoryForm extends Component<
    BaseCategoryFormProps & WithStyles<"cardActions" | "textField">,
    BaseCategoryFormState
  > {
    state = {
      formData: {
        description: this.props.description,
        name: this.props.name
      }
    };

    handleInputChange = event => {
      const { target } = event;
      this.setState(prevState => ({
        formData: { ...prevState.formData, [target.name]: target.value }
      }));
    };

    render() {
      const {
        classes,
        confirmButtonLabel,
        description,
        handleConfirm,
        name,
        title,
        errors
      } = this.props;
      const errorList = errors.reduce((acc, curr) => {
        acc[curr.field] = curr.message;
        return acc;
      }, {});
      console.log(errorList);
      return (
        <Card>
          <CardContent>
            <Typography variant="display1">{title}</Typography>
          </CardContent>
          <CardContent>
            <TextField
              autoFocus
              name="name"
              label={i18n.t("Name", { context: "category" })}
              defaultValue={name}
              className={classes.textField}
              onChange={this.handleInputChange}
              error={Boolean(errorList["name"])}
              helperText={errorList["name"]}
            />
            <TextField
              name="description"
              label={i18n.t("Description (optional)")}
              defaultValue={description}
              multiline
              onChange={this.handleInputChange}
              error={Boolean(errorList["description"])}
              helperText={errorList["description"]}
            />
          </CardContent>
          <CardContent>
            <CardActions className={classes.cardActions}>
              <Button
                variant="raised"
                color="primary"
                onClick={() => handleConfirm(this.state.formData)}
              >
                {confirmButtonLabel}
              </Button>
              <Button color="primary" onClick={() => window.history.back()}>
                {i18n.t("Cancel", { context: "button" })}
              </Button>
            </CardActions>
          </CardContent>
        </Card>
      );
    }
  }
);
