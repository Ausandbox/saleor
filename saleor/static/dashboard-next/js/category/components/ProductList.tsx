import * as React from "react";
import Avatar from "material-ui/Avatar";
import Button from "material-ui/Button";
import List, {
  ListItemAvatar,
  ListItem,
  ListItemText,
  ListSubheader
} from "material-ui/List";
import Typography from "material-ui/Typography";
import Cached from "material-ui-icons/Cached";
import MoreVert from "material-ui-icons/MoreVert";
import { Link } from "react-router-dom";

import { CategoryPropertiesQuery } from "../gql-types";
import { categoryAddUrl } from "../";
import i18n from "../../i18n";
import Skeleton from "../../components/Skeleton";

interface ProductListProps {
  products?: CategoryPropertiesQuery["category"]["products"]["edges"];
  handleLoadMore();
  canLoadMore: boolean;
}

export const ProductList: React.StatelessComponent<ProductListProps> = ({
  products,
  handleLoadMore,
  canLoadMore
}) => (
  <List>
    {products === undefined ? (
      <ListItem>
        <ListItemAvatar>
          <Avatar>
            <Cached />
          </Avatar>
        </ListItemAvatar>
        <ListItemText>
          <Skeleton />
        </ListItemText>
      </ListItem>
    ) : products.length > 0 ? (
      products.map(edge => (
        <ListItem
          button
          key={edge.node.id}
          component={props => <Link to="#" {...props} />}
        >
          <ListItemAvatar>
            <Avatar src={edge.node.thumbnailUrl} />
          </ListItemAvatar>
          <ListItemText primary={edge.node.name} />
        </ListItem>
      ))
    ) : (
      <ListSubheader>{i18n.t("No products found")}</ListSubheader>
    )}
    {canLoadMore && (
      <ListItem button key="more" onClick={handleLoadMore}>
        <ListItemAvatar>
          <Avatar>
            <MoreVert />
          </Avatar>
        </ListItemAvatar>
        <ListItemText primary={i18n.t("Load more", { context: "button" })} />
      </ListItem>
    )}
  </List>
);

export default ProductList;
