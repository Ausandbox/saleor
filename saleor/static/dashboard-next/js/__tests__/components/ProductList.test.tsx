import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import { ProductList } from "../../category/components/ProductList";
import productListFixture from "./fixtures/productList";

describe("<ProductList />", () => {
  it("renders while data is loading", () => {
    const component = renderer.create(
      <MemoryRouter>
        <ProductList
          loading={true}
          canLoadMore={false}
          handleLoadMore={jest.fn()}
          products={[]}
        />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders properly when data is loaded", () => {
    const component = renderer.create(
      <MemoryRouter>
        <ProductList
          loading={false}
          canLoadMore={false}
          handleLoadMore={jest.fn()}
          products={productListFixture}
        />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders properly 'load more' button", () => {
    const component = renderer.create(
      <MemoryRouter>
        <ProductList
          loading={false}
          canLoadMore={true}
          handleLoadMore={jest.fn()}
          products={productListFixture}
        />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders properly when product list is empty", () => {
    const component = renderer.create(
      <MemoryRouter>
        <ProductList
          loading={false}
          canLoadMore={false}
          handleLoadMore={jest.fn()}
          products={[]}
        />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
