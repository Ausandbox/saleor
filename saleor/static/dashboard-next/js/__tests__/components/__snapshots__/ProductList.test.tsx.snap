// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> renders properly 'load more' button 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_ZoF6Xi8-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Lewis PLC
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_07Fa6v8-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Kennedy-Ramirez
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_zeczDly-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Newman, Ashley and Roberson
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_VX2ikxl-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Jacobson, Stewart and Rodriguez
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_wl4FUo7-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Sloan Ltd
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_c0nSiAe-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Foster-Beard
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_qHI83dZ-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Henson PLC
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_06TblXK-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Novak, Lopez and Cook
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_yMYqzTc-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Thompson-Jones
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_eWtYYcz-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Cunningham, Torres and Day
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <div
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    onBlur={[Function]}
    onClick={[MockFunction]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17 MuiAvatar-colorDefault-18"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-20"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Load more
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </div>
</ul>
`;

exports[`<ProductList /> renders properly when data is loaded 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_ZoF6Xi8-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Lewis PLC
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_07Fa6v8-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Kennedy-Ramirez
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_zeczDly-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Newman, Ashley and Roberson
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_VX2ikxl-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Jacobson, Stewart and Rodriguez
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_wl4FUo7-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Sloan Ltd
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_c0nSiAe-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Foster-Beard
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_qHI83dZ-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Henson PLC
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_06TblXK-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Novak, Lopez and Cook
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/6_yMYqzTc-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Thompson-Jones
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
  <a
    className="MuiButtonBase-root-57 MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12 MuiListItem-button-13"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-17"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-19"
        sizes={undefined}
        src="/media/__sized__/products/saleor/static/placeholders/t-shirts/5_eWtYYcz-crop-c0-5__0-5-255x255-70.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        Cunningham, Torres and Day
      </h3>
    </div>
    <span
      className="MuiTouchRipple-root-60"
    />
  </a>
</ul>
`;

exports[`<ProductList /> renders properly when product list is empty 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <li
    className="MuiListSubheader-root-67 MuiListSubheader-sticky-71"
  >
    No products found
  </li>
</ul>
`;

exports[`<ProductList /> renders while data is loading 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <li
    className="MuiListItem-root-5 MuiListItem-dense-9 MuiListItem-gutters-12"
    disabled={false}
  >
    <div
      className="MuiAvatar-root-17 MuiAvatar-colorDefault-18"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-20"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 8l-4 4h3c0 3.31-2.69 6-6 6-1.01 0-1.97-.25-2.8-.7l-1.46 1.46C8.97 19.54 10.43 20 12 20c4.42 0 8-3.58 8-8h3l-4-4zM6 12c0-3.31 2.69-6 6-6 1.01 0 1.97.25 2.8.7l1.46-1.46C15.03 4.46 13.57 4 12 4c-4.42 0-8 3.58-8 8H1l4 4 4-4H6z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root-26"
    >
      <h3
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiListItemText-primary-29"
      >
        <span
          className="Component-skeleton-56"
          style={undefined}
        >
          ‌
        </span>
      </h3>
    </div>
  </li>
</ul>
`;
